### ISSUE TITLE
Cannot get past bad_server_salt

### ISSUE THREAD

**Issue by Fnux** [Fnux/telegram-mt-elixir#6]
Commit dcc40baf51a23d93b73c1f24767620ba749a17d3 automatically update the server salt when `bad_server_salt` is received. However, the next messages are still received a `bad_server_salt`.

 > 48: incorrect server salt (in this case, the bad_server_salt response is received with the correct salt, and the message is to be re-sent with it)

```
{2830428717143823206,
 %{bad_msg_id: 6446267035327397888, bad_msg_seqno: 47, error_code: 48,
   msg_id: 6446267039641415681, name: "bad_server_salt",
   new_server_salt: -1188075740232232665}}
```

**Comment by Fnux** [comment]
Endianness issue, should be fixed soon.

**Comment by Fnux** [comment]
Fixed by 141ebf4e5047eb2e74fdff541f199334d6d70e5b, waiting for merge.

**Comment by Fnux** [comment]
#12 : merged.
